test/CMakeLists.txt

```diff
@@ -22,7 +26,7 @@ CPMAddPackage(
 )
 
 if(TEST_INSTALLED_VERSION)
-  find_package(Greeter REQUIRED)
+  find_package(Greeter ${PROJECT_VERSION} REQUIRED)
```

Comment:
The `TEST_INSTALLED_VERSION` flag should be used either in CI or against a custom root directory for testing, so the version tag isn't needed, as no other will be present. I agree that this could be documented here.

Reply:
How would you test the compatibility of a real dynamic library?

 `# TODO: COMPATIBILITY SameMajorVersion`

see https://crascit.com/2019/10/16/cppcon-2019-deep-cmake-for-library-authors/

Reply:
This would be defined by the `COMPATIBILITY` flag of PackageProject.cmake?

Reply:
IMHO, it is the decision of an **end-user** to look for **any** version, or use `find_package(<PackageName>  1.2.3 EXACT ...)` version, or a compatible version.

In our case, the test failed if not EXACT is used!

Reply:
Still, if used as intended: a unit test for a clean system / root directory, there will be only one version of the project installed. Also currently I don't think that `PROJECT_VERSION` is properly defined or guaranteed to be the same as defined in the outer CMakeLists.

Reply:
yes, that is true and I fixed it